src/processors/transform.rs

```diff
@@ -0,0 +1,498 @@
+}
+
+impl Default for Transform {
+    fn default() -> Self {
+        Transform([[1.0, 0.0, 0.0, 0.0],
+                   [0.0, 1.0, 0.0, 0.0],
+                   [0.0, 0.0, 1.0, 0.0],
+                   [0.0, 0.0, 0.0, 1.0]])
+    }
+}
+
+impl From<[[f32; 4]; 4]> for Transform {
+    fn from(matrix: [[f32; 4]; 4]) -> Self {
+        Transform(matrix)
+    }
+}
+
+impl Into<[[f32; 4]; 4]> for Transform {
+    fn into(self) -> [[f32; 4]; 4] {
+        self.0
+    }
+}
+
+/// Initialization flag.
+/// Added to entity with a `LocalTransform` component after the first update.
+#[derive(Default, Copy, Clone)]
+pub struct Init;
+impl Component for Init {
+    type Storage = NullStorage<Init>;
+}
+
+/// Component for defining a parent entity.
+pub struct Child {
+    /// The parent entity
+    parent: Entity,
+
+    /// Flag for whether the child was changed
+    dirty: AtomicBool,
+}
+
+impl Child {
+    pub fn new(entity: Entity) -> Child {
+        Child {
+            parent: entity,
+            dirty: AtomicBool::new(true),
+        }
+    }
+
+    #[inline]
+    pub fn parent(&self) -> Entity {
+        self.parent
+    }
+    #[inline]
+    pub fn set_parent(&mut self, entity: Entity) {
+        self.parent = entity;
+        self.flag(true);
+    }
+
+    /// Flag that parent has been changed
+    ///
+    /// Note: `set_parent` flags the parent.
+    #[inline]
+    pub fn flag(&self, dirty: bool) {
+        self.dirty.store(dirty, Ordering::SeqCst);
+    }
+
+    /// Returns whether the parent was changed.
+    #[inline]
+    pub fn is_dirty(&self) -> bool {
+        self.dirty.load(Ordering::SeqCst)
+    }
+}
+
+impl Component for Child {
+    type Storage = VecStorage<Child>;
+}
+
+/// Transformation processor.
+///
+/// Handles updating `Transform` components based on the `LocalTransform` component and parents.
+pub struct TransformProcessor {
+    // Map of entities to index in sorted vec.
+    indices: HashMap<Entity, usize>,
+
+    // Vec of entities with parents before children. Only contains entities with parents.
+    sorted: Vec<(Entity, Entity)>,
+
+    // New entities in current update
+    new: Vec<Entity>,
+
+    // Entities that have been removed in current frame.
+    dead: HashSet<Entity>,
+
+    // Child entities that were dirty.
+    dirty: HashSet<Entity>,
+
+    // Prevent circular infinite loops with parents.
+    swapped: HashSet<Entity>,
+}
+
+impl TransformProcessor {
+    pub fn new() -> TransformProcessor {
+        TransformProcessor {
+            indices: HashMap::new(),
+            sorted: Vec::new(),
+            new: Vec::new(),
+            dead: HashSet::new(),
+            dirty: HashSet::new(),
+            swapped: HashSet::new(),
+        }
+    }
+}
+
+impl Processor<Arc<Mutex<Context>>> for TransformProcessor {
+    fn run(&mut self, arg: RunArg, _: Arc<Mutex<Context>>) {
+        // Fetch world and gets entities/components
+        let (locals, mut globals, mut init, parents) = arg.fetch(|w| {
```

Comment:
I think variable names also should be changed, so `parents` should be `children` (since you are iterating over all `Child` components), `parent.parent()` should be `child.parent()` (since you are accessing the parent of a `Child`), and so on.